server/src/service.rs

```diff
@@ -458,4 +474,42 @@ impl Service {
 
         Ok(signal_emitter)
     }
+
+    pub async fn remove_client(&self, old_owner: &str) -> Option<OwnedObjectPath> {
+        self.clients.lock().await.remove(old_owner)
+    }
+
+    pub async fn cleanup_sessions_on_client_disconnect(&self) -> Result<(), Error> {
+        let rule = zbus::MatchRule::builder()
+            .msg_type(zbus::message::Type::Signal)
+            .sender("org.freedesktop.DBus")?
+            .member("NameOwnerChanged")?
+            .build();
```

Comment:
This is too broad, you will get too many NameOwnerChanged, not only for secrets service only.

Reply:
by using add_arg no ?